tests/core/data/test_base_viz.py

```diff
@@ -0,0 +1,186 @@
+from typing import Any, List, Sequence
+
+import numpy as np
+import pytest
+import torch
+from pytorch_lightning import seed_everything
+from pytorch_lightning.trainer.states import RunningStage
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+from flash.core.data.base_viz import BaseVisualization
+from flash.core.data.data_source import DefaultDataKeys
+from flash.core.data.utils import _CALLBACK_FUNCS, _STAGES_PREFIX
+from flash.core.utilities.imports import _IMAGE_AVAILABLE
+from flash.image import ImageClassificationData
+
+if _IMAGE_AVAILABLE:
+    from PIL import Image
+
+
+def _rand_image():
+    return Image.fromarray(np.random.randint(0, 255, (196, 196, 3), dtype="uint8"))
+
+
+class CustomBaseVisualization(BaseVisualization):
+
+    def __init__(self):
+        super().__init__()
+
+        self.show_load_sample_called = False
+        self.show_pre_tensor_transform_called = False
+        self.show_to_tensor_transform_called = False
+        self.show_post_tensor_transform_called = False
+        self.show_collate_called = False
+        self.per_batch_transform_called = False
+
+    def show_load_sample(self, samples: List[Any], running_stage: RunningStage):
+        self.show_load_sample_called = True
+
+    def show_pre_tensor_transform(self, samples: List[Any], running_stage: RunningStage):
+        self.show_pre_tensor_transform_called = True
+
+    def show_to_tensor_transform(self, samples: List[Any], running_stage: RunningStage):
+        self.show_to_tensor_transform_called = True
+
+    def show_post_tensor_transform(self, samples: List[Any], running_stage: RunningStage):
+        self.show_post_tensor_transform_called = True
+
+    def show_collate(self, batch: Sequence, running_stage: RunningStage) -> None:
+        self.show_collate_called = True
+
+    def show_per_batch_transform(self, batch: Sequence, running_stage: RunningStage) -> None:
+        self.per_batch_transform_called = True
+
+    def check_reset(self):
+        self.show_load_sample_called = False
+        self.show_pre_tensor_transform_called = False
+        self.show_to_tensor_transform_called = False
+        self.show_post_tensor_transform_called = False
+        self.show_collate_called = False
+        self.per_batch_transform_called = False
+
+
+@pytest.mark.skipif(not _IMAGE_AVAILABLE, reason="image libraries aren't installed.")
+class TestBaseViz:
+
+    def test_base_viz(self, tmpdir):
+
+        seed_everything(42)
+        tmpdir = Path(tmpdir)
+
+        train_images = [str(tmpdir / "a1.png"), str(tmpdir / "b1.png")]
+
+        _rand_image().save(train_images[0])
+        _rand_image().save(train_images[1])
+
+        class CustomImageClassificationData(ImageClassificationData):
+
+            @staticmethod
+            def configure_data_fetcher(*args, **kwargs) -> CustomBaseVisualization:
+                return CustomBaseVisualization(*args, **kwargs)
+
+        B: int = 2  # batch_size
```

Comment:
```suggestion
        BS: int = 2  # batch_size
```
alt least more then one letter